pennylane/measurements/shots.py

```diff
@@ -161,6 +161,20 @@ def __deepcopy__(self, memo):
         memo[id(self)] = self
         return self
 
+    def __str__(self):
+        """The string representation of the class"""
+
```

Comment:
One option: if the shot vector is trivial, we can leave it out in the string. Makes it a little nicer to read:
```
if not self.has_partitioned_shots:
    return f"Shots(total={self.total_shots})"
```